docs/WritingTests.asciidoc

```diff
@@ -1248,3 +1251,48 @@ the clone source job or specifying the variable explicitly:
 ----
 clone_job.pl --from https://openqa.opensuse.org  --host localhost 24 TESTDEBUG=1 SKIPTO=consoletest-yast2_i
 ----
+
+=== Defining a custom test schedule or custom test modules
+
+Normally the test schedule, that is which test modules should be executed and
+which order, is prescribed by the +main.pm+ file within the test distribution.
+Additionally it is possible to exclude certain test modules from execution
+using the os-autoinst test variables +INCLUDE_MODULES+ and +EXCLUDE_MODULES+ as
+well as define a custom schedule using the test variable +SCHEDULE+.
+Also test modules can be defined and overridden on-the-fly using a
+downloadable asset.
+
+For example one can schedule a job on a production instance with a custom
+schedule consisting of two modules from the provided test distribution plus
+one test module which is defined dynamically and downloaded as an asset from
+an external trusted download domain:
+
+----
+openqa-clone-job --from https://openqa.opensuse.org --host https://openqa.opensuse.org 24 SCHEDULE=tests/boot/boot_to_desktop,tests/console/consoletest_setup,foo ASSET_1_URL=https://example.org/my/test/foo.pm
+----
+
+where "foo.pm" is a file in test module format, e.g. the following content:
+
+----
+use base 'consoletest';
+use strict;
+use testapi;
+
+sub run {
+    select_console 'root-console';
+    assert_script_run 'foo';
+}
+
+sub post_run_hook {}
+1;
+----
+
+This could be used e.g. in bug investigations, trying out new test modules
+which are hard to test locally, etc. I assume documentation is hard to find for
```

Comment:
I've read it 3 times, still reads a bit weird, can you improve the wording? I'll merge after, unless somebody else merges ;)

Reply:
rephrased, please check again